src/analysis.zig

```diff
@@ -5873,6 +5873,7 @@ pub fn lookupSymbolFieldInit(
         const unwrapped =
             try analyser.resolveUnwrapErrorUnionType(container_type, .payload) orelse
             try analyser.resolveOptionalUnwrap(container_type) orelse
+            try analyser.resolveDerefType(container_type) orelse
             break;
         container_type = unwrapped;
     }
```

Comment:
We can simplify this by using `resolveDeclLiteralResultType` which does exactly what is needed for decl literals. The entire while loop can be replaced with the following code:
```zig
container_type = try container_type
    .resolveDeclLiteralResultType()
    .instanceTypeVal(analyser) orelse container_type;
```